src/WinRT.Runtime/Marshalers.cs

```diff
@@ -41,37 +114,39 @@ public void Dispose()
             _header = IntPtr.Zero;
         }
 
-        public static unsafe MarshalString CreateMarshaler(string value)
+        public static MarshalString CreateMarshaler(string value)
         {
-            if (value == null) return null;
-
-            var m = new MarshalString();
+            return string.IsNullOrEmpty(value) ? null : new MarshalString(value);
+        }
 
+        public unsafe IntPtr GetAbi()
+        {
             bool success = false;
             try

```

Comment:
Do you actually need try/finally in this method? I would expect the cleanup to be taken care of by the calling code if something goes wrong here.

Reply:
If something went wrong and the cleanup in this method actually kicked in, I think you would actually ended with double-free of the GCHandle due to buggy implementation of `MarshalString.Dispose`. `MarshalString.Dispose` disposes the GCHandle, but it does not clear the GCHandle value, so calling `MarshalString.Dispose` twice is going to free the GCHandle twice.

Reply:
right, in this case it's redundant